### ISSUE TITLE
VS crash due to ArgumentOutOfRangeException thrown by incremental parser

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#659]
Stack trace:

```
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode.ChildThatContainsPosition(Int32 position)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode.FindTokenInternal(Int32 position)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode.FindToken(Int32 position, Boolean findInsideTrivia)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender.ExtendToAffectedRange(CSharpSyntaxNode oldTree, TextChangeRange changeRange)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender..ctor(Lexer lexer, CSharpSyntaxNode oldTree, IEnumerable`1 changes)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.SyntaxParser..ctor(Lexer lexer, LexerMode mode, CSharpSyntaxNode oldTree, IEnumerable`1 changes, Boolean allowModeReset, Boolean preLexIfNotIncremental, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser..ctor(Lexer lexer, CSharpSyntaxNode oldTree, IEnumerable`1 changes, LexerMode lexerMode, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTree.WithChanges(SourceText newText, IReadOnlyList`1 changes)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTree.WithChangedText(SourceText newText)
   at Microsoft.CodeAnalysis.DocumentState.<IncrementallyParseTreeAsync>d__9.MoveNext()
```

Crash dump stored on internal share tomat\bugs\659. It isn't that much useful though since the stack is gone. I have added exception filter to IncrementallyParseTreeAsync so we can get a better stack next time.

Repro:
Extract tomat\bugs\659\Roslyn.zip, open AbstractEditAndContinueAnalyzer.cs, go to line 1059

var lambdaBodyMatch = ComputeLambdaBodyMatch(oldBody1, newLambda, activeNodes, lazyActiveLambdas, trackingService, diagnostics);

And delete "trackingService" identifier.

**Comment by pharring** [comment]
@tmat Can you remember what you were doing at the time?

**Comment by tmat** [comment]
Yes, updated the description above with 100% repro.

**Comment by gafter** [comment]
This looks exactly like the issue that @mattwar is working on.

**Comment by mattwar** [comment]
This is the same as #605 and fixed by #736
